File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/AddressService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbAddress;							  
import entity.PageResult;
import entity.Result;
/**
 * address服务层接口
 * @author jieweili
 *
 */
public interface AddressService {

    /**
     * 返回全部列表(已弃置)
     * @return
     */
    List<TbAddress> findAll();
 
    /**
     * 返回分页对象(过时)
     * @param pageNum
     * @param pageSize
     * @return
     */
	 
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加数据
     * @param address
     * @return
     */
    public Result saveAddress(TbAddress address);

    /**
     * 根据id查询一个对象
     * @param id
     * @return
     */
    TbAddress getAddressById(Long id);

    /**
     * 修改对象
     * @param address
     * @return
     */
    Result updateAddress(TbAddress address);

    /**
     * 批量删除
     * @param idList
     * @return
     */
    Result removeAddress(Long[] idList);

    /**
     * 根据条件分页查询
     * @param address
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult listAddressByExample(TbAddress address, Integer pageNum, Integer pageSize);
 
}

File: pinyougou_parent/pinyougou_manager_web/src/main/webapp/js/service/addressService.js
app.service('addressService', function ($http) {
    //旧的查询方法(弃置)								   
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../address/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize)
    };

    //查询方法,读取列表数据绑定到表单中
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../address/findPage.do?pageNum=' + pageNum + "&pageSize=" + pageSize, searchEntity)
    };

    //删除方法
    this.removeAddressList = function (idList) {
        return $http.get('../address/removeAddress.do?idList=' + idList)
    };

    //根据id查询
    this.findOne = function (id) {
        return $http.get('../address/findOne.do?id=' + id)
    };

    //修改品牌
    this.updateAddress = function (address) {
        return $http.post('../address/updateAddress.do', address)
    };

    //新增品牌
    this.saveAddress = function (address) {
        return $http.post('../address/saveAddress.do', address)
    };   
});
